### PR TITLE
Test AbstractSource read behavior

### PR DESCRIPTION
**Pull request by sherifnada**
## What
Tests read behavior and state management for full refresh/incremental reads with different slice/checkpoint interval configurations

### CODE REVIEW

**Comment by davinchia** [comment]
@sherifnada sorry for the extra files. I was cleaning this up and merged master in. 

I made some slight edits to one test to get it to pass. The TLDR is that mocker needs to be configured differently for property mocks..

This looks good. I'd want to implement the commented test case before merging.

What do you think about a test case for an incremental stream with checkpoint and slicing? Not necessary?

**Review by davinchia — APPROVED** [review]

File: airbyte-cdk/python/unit_tests/base_python/test_abstract_source.py [hunk 1]
```diff
@@ -63,34 +114,173 @@ def test_discover(mocker):
...
+def test_valid_incremental_read_with_checkpoint_interval(mocker, logger):
+    stream_output = [{"k1": "v1"}, {"k2": "v2"}]
+    s1 = MockStream([({"sync_mode": SyncMode.incremental, "stream_state": {}}, stream_output)], name="s1")
+    s2 = MockStream([({"sync_mode": SyncMode.incremental, "stream_state": {}}, stream_output)], name="s2")
+    state = {"cursor": "value"}
+    mocker.patch.object(MockStream, "get_updated_state", return_value=state)
+    mocker.patch.object(MockStream, "supports_incremental", return_value=True)
+    mocker.patch.object(MockStream, "get_json_schema", return_value={})
+    # Tell the source to output one state message per record
+    mocker.patch.object(MockStream, "state_checkpoint_interval", new_callable=mocker.PropertyMock, return_value=1)
```
Comment (davinchia) [inline airbyte-cdk/python/unit_tests/base_python/test_abstract_source.py:205]: @sherifnada this `PropertyMock` was what was missing.

Comment (sherifnada) [reply airbyte-cdk/python/unit_tests/base_python/test_abstract_source.py:205]: thanks boss

**Comment by sherifnada** [comment]
@davinchia 
> I'd want to implement the commented test case before merging.
done https://github.com/airbytehq/airbyte/pull/3285

> Can we also add docstring on each test method with a brief description of what is going on?
done

> What do you think about a test case for an incremental stream with checkpoint and slicing? Not necessary?
done

**Pull request merged**